### ISSUE TITLE
CGContextRestoreGState: invalid context

### ISSUE THREAD

**Issue by robmcmullen** [robmcmullen/peppy#552]
**[reported by askadar@...]** _[Trac time 20081011 025955Z]_ I'm testing peppy on OS X 10.5 using Python 2.6. 

I created a new buffer by clicking the new document button. On every key stroke, peppy reports the following error:

```
Fri Oct 10 22:55:58 Serenity.local Python[63504] <Error>: CGContextRestoreGState: invalid context
Fri Oct 10 22:55:58 Serenity.local Python[63504] <Error>: CGContextRestoreGState: invalid context
Fri Oct 10 22:55:58 Serenity.local Python[63504] <Error>: CGContextRestoreGState: invalid context
Fri Oct 10 22:55:58 Serenity.local Python[63504] <Error>: CGContextRestoreGState: invalid context
Fri Oct 10 22:55:59 Serenity.local Python[63504] <Error>: CGContextRestoreGState: invalid context
Fri Oct 10 22:56:00 Serenity.local Python[63504] <Error>: CGContextRestoreGState: invalid context
Fri Oct 10 22:56:01 Serenity.local Python[63504] <Error>: CGContextRestoreGState: invalid context
Fri Oct 10 22:56:02 Serenity.local Python[63504] <Error>: CGContextRestoreGState: invalid context
Fri Oct 10 22:56:02 Serenity.local Python[63504] <Error>: CGContextRestoreGState: invalid context
Fri Oct 10 22:56:14 Serenity.local Python[63504] <Error>: CGContextRestoreGState: invalid context
....
```

**Comment by robmcmullen** [comment]
_[Trac time 20081013 211450Z]_ Thanks for the report.  I had borrowed a mac for some testing but currently don't access to it...  Hopefully I'll be getting a mac soon -- I am waiting for tomorrow's announcement about the new laptops before buying one. :)

It looks like it might be related to http://trac.wxwidgets.org/ticket/4272 which is a problem with the statusbar updating.  It says that using a CallAfter worked around the problem, so I wonder if changing line 615 in peppy/fundamental.py from

```
        self.status_info.setText("L%d C%d F%d S%d %d"  ...
```

to

```
        wx.CallAfter(self.status_info.setText, "L%d C%d F%d S%d %d" ...
```

will prevent the problem?

**Comment by robmcmullen** [comment]
**[comment by askadar@...]** _[Trac time 20081013 212322Z]_ That fixes it; no more error messages on the console.

**Comment by robmcmullen** [comment]
_[Trac time 20081013 214848Z]_ Thanks for testing it.  I'll put the fix into svn.

**Comment by robmcmullen** [comment]
_[Trac time 20081013 220541Z]_ 595284cc4781bb3e6f6cec5cdb503b760d6bfc08 Fixed #552: Workaround for CGContextRestoreGState errors on mac when setting status bar inside event handler
- currently just a workaround using wx.CallAfter; a better solution might be to rework the cursor position reporting into the idle event handler, but it will probably have to wait until after I get a Mac.
